src/mlpack/tests/hmm_test.cpp

```diff
@@ -1228,5 +1229,598 @@ BOOST_AUTO_TEST_CASE(DiscreteHMMLoadSaveTest)
           hmm2.Emission()[j].Probabilities()[i], 1e-3);
 }
 
-BOOST_AUTO_TEST_SUITE_END();
+/********************************************/
+/** DiagonalGMM Hidden Markov Models Tests **/
+/********************************************/
+
+//! Make sure the prediction of DiagonalGMM HMMs is reasonable.
+BOOST_AUTO_TEST_CASE(DiagonalGMMHMMPredictTest)
+{
+  // This test is probabilistic, so we perform it three times to make it robust.
+  bool success = false;
+  for (size_t trial = 0; trial < 3; trial++)
+  {
+    std::vector<DiagonalGMM> gmms(2);
+    gmms[0] = DiagonalGMM(2, 2);
+
+    gmms[0].Component(0) = DiagonalGaussianDistribution("3.25 2.10",
+        "0.97 1.00");
+    gmms[0].Component(1) = DiagonalGaussianDistribution("5.03 7.28",
+        "1.20 0.89");
+
+    gmms[1] = DiagonalGMM(3, 2);
+    gmms[1].Weights() = arma::vec("0.3 0.2 0.5");
+    gmms[1].Component(0) = DiagonalGaussianDistribution("-2.48 -3.02",
+        "1.02 0.80");
+    gmms[1].Component(0) = DiagonalGaussianDistribution("-1.24 -2.40",
+        "0.85 0.78");
+    gmms[1].Component(0) = DiagonalGaussianDistribution("-5.68 -4.83",
+        "1.42 0.96");
+
+    // Initial probabilities.
+    arma::vec initial("1 0");
+
+    // Transition matrix.
+    arma::mat transProb("0.40 0.70;"
+                        "0.60 0.30");
+
+    // Build the model.
+    HMM<DiagonalGMM> hmm(initial, transProb, gmms);
+
+    // Make a sequence of observations according to transition probabilities.
+    arma::mat observations(2, 1000);
+    arma::Row<size_t> states(1000);
+
+    // Set initial state to zero.
+    states[0] = 0;
+    observations.col(0) = gmms[0].Random();
+
+    for (size_t i = 1; i < 1000; i++)
+    {
+      double randValue = math::Random();
+
+      if (randValue <= transProb(0, states[i - 1]))
+        states[i] = 0;
+      else
+        states[i] = 1;
+
+      observations.col(i) = gmms[states[i]].Random();
+    }
+
+    // Predict the most probable hidden state sequence.
+    arma::Row<size_t> predictions;
+    hmm.Predict(observations, predictions);
+
+    // Check them.
+    success = true;
+    for (size_t i = 0; i < 1000; i++)
+    {
+      if (predictions[i] != states[i])
+      {
+        success = false;
+        break;
+      }
+    }
+
+    if (success)
```

Comment:
Looks like this test is missing a ``BOOST_REQUIRE_EQUAL(success, true);`` check at the end.

Reply:
Oh.. Thanks!!